KeyViewLoops.md

```diff
@@ -0,0 +1,49 @@
+# Key View Loops
+
+### Convention
+It is recommended to let the window automatically calculate your key view loop for you. There are very few cases in which the automatically calculated key view loop is not sufficient.
+
+### Rationale
+Allowing the window to automatically calculate the key view loop (by setting `window.autoRecalculatesKeyViewLoop = true`), you can:
+- Better conform to the user's expectation of the key view loop laid out in geometric order (left to right, top to bottom for the english locale at the time of writing)
+- Protect against future OS updates where Apple may change the behavior of the key-view loop.
+- Save yourself a lot of hassle of manually managing a key view loop
+
+One thing to note: setting `autoRecalculatesKeyViewLoop` to `true` means that any NSView whose `nextKeyView` property you have manually set will be ignored. This is seemingly because AppKit will constantly recalculate the key view loop every time a view is added or the user tabs to a new view, stamping over any customizations to the key view loop done by  the developer.
+
+Manually managing your own key-view loop (by setting `window.autoRecalculatesKeyViewLoop = false`) has a few major downsides:
+- Doing so means you will have no key view loop except for whichever views you have manually set the `nextKeyView` property of in your project.
+- Any call to `window.recalculateKeyViewLoop()` will stamp over your customizations, as we have already described.
+
+
+### Exceptions
+If your window has a fairly static view heirarchy it is easier to set the nextKeyView properties for every applicable view. If you only have a few specific views whose `nextKeyView` property you want to modify, you can call `window.recalculateKeyViewLoop()` once to get an AppKit generated key view loop, and then set up your custom key views.
+
+
+### Examples
```

Comment:
I still think it would be good to add at least one example where someone can fix a problem while retaining an automatic loop and shouldn't switch to manual loops. Put another way - the rest of this page is an awesome argument against using manual key view loops. But then the example is "Here is a rough overview of what your code may look like if you manually set up your own key view.". Maybe the answer is just better phrasing of that as something like "here's the likely problem you would face by manually setting up your own key view"?

Reply:
Yeah sorry, I hadn't thought of one up until now, which is also that is the one comment I have left unresolved. I haven't set aside time to come up with a strong example just yet.